Update changelogs for Docker Desktop edge 2.4.2.0 (#11550)
docker/docker.github.io: docker-for-windows/edge-release-notes.md

For information about Stable releases, see the [Stable release notes](release-notes.md). For Docker Desktop system requirements, see
[What to know before you install](install.md#what-to-know-before-you-install).

## Docker Desktop Community 2.4.2.0
2020-10-15

### Upgrades

- [Docker Engine 20.10.0 beta1](https://github.com/docker/docker-ce/blob/0fc7084265b3786a5867ec311d3f916af7bf7a23/CHANGELOG.md)
- [Docker Compose CLI - 0.1.22](https://github.com/docker/compose-cli/releases/tag/v0.1.22)
- [Linux kernel 5.4.39](https://hub.docker.com/layers/linuxkit/kernel/5.4.39-f39f83d0d475b274938c86eaa796022bfc7063d2/images/sha256-8614670219aca0bb276d4749e479591b60cd348abc770ac9ecd09ee4c1575405?context=explore)
- [Kubernetes 1.19.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.19.2)
- [Go 1.15.2](https://github.com/golang/go/issues?q=milestone:Go1.15.2+label:CherryPickApproved)

* Deprecation
  - Docker Desktop cannot be installed on Windows 1703 (build 15063) anymore.

### Bug fixes and minor changes

> 

Reviewer: I think these grpcfuse changes are relevant on Windows: (although no-one complained about them so the impact might be less)

```suggestion

- When sharing a file into a container (e.g. `docker run -v C:\Users\...\.gitconfig`) don't watch the parent directory. Fixes [docker/for-mac#4981](https://github.com/docker/for-mac/issues/4981).
- Always flush filesystem caches synchronously on container start. See [docker/for-mac#4943](https://github.com/docker/for-mac/issues/4943).
```

> - Fix an issue that intermittently make the WSL 2 backend fail to start.

Reviewer 2: ```suggestion
- Fixed an issue that can intermittently cause the WSL 2 backend to fail to start.
```

> - Fix NFS mounting, see [docker/for-mac#4958](https://github.com/docker/for-mac/issues/4958).

Reviewer 2: ```suggestion
- Fixed an issue related to NFS mounting. See [docker/for-mac#4958](https://github.com/docker/for-mac/issues/4958).
```

> - Fix an issue with http proxy exclude lists containing entries shuch as localhost or 127.0.0.1, fixes [docker/for-win#8750](https://github.com/docker/for-win/issues/8750).

Reviewer 2: ```suggestion
- Fixed an issue with http proxy exclude lists containing entries such as `localhost` or `127.0.0.1`. Fixes [docker/for-win#8750](https://github.com/docker/for-win/issues/8750).
```

> - Fix an issue with http proxy exclude lists containing entries shuch as localhost or 127.0.0.1, fixes [docker/for-win#8750](https://github.com/docker/for-win/issues/8750).

Reviewer 2: ```suggestion
- Fixed an issue with http proxy exclude lists containing entries such as `localhost` or `127.0.0.1`. Fixes [docker/for-win#8750](https://github.com/docker/for-win/issues/8750).
```

- When WSL integration process unexpectedly stops, the user is now notified and can decide to restart it or not, instead of always try to restart it in a loop.

> - Fix container logs lagging under heavy load. Fixes [docker/for-win#8216](https://github.com/docker/for-win/issues/8216).

Reviewer 2: ```suggestion
- Fixed an issue related to container logs that lag under heavy load. Fixes [docker/for-win#8216](https://github.com/docker/for-win/issues/8216).
```
